app/controllers/repo_based_controller.rb

```diff
@@ -1,11 +1,15 @@
 class RepoBasedController < ApplicationController
   protected
 
-    def name_from_params(options)
-      [options[:name], options[:format]].compact.join('.')
-    end
+  def name_from_params(options)
+    [options[:name], options[:format]].compact.join('.')
+  end
 
-    def find_repo(options)
-      Repo.find_by_full_name(options[:full_name])
+  def find_repo(options)
+    if repo = Repo.find_by_full_name(options[:full_name])
+      repo.update_repo_info! if repo.updated_at < (Time.now - 6.hours)
```

Comment:
I don't think it should be present in We can add a cron to update the repos after every X hours.